template/java8/entrypoint/src/main/java/com/openfaas/entrypoint/App.java

```diff
@@ -72,12 +62,11 @@ public void handle(HttpExchange t) throws IOException {
             //     System.out.println("Req header " + entry.getKey() + " " + entry.getValue());
             // }
 
-            IRequest req = new Request(requestBody, reqHeadersMap,t.getRequestURI().getRawQuery(), t.getRequestURI().getPath());
+            IRequest req = new Request(reqHeadersMap,t.getRequestURI().getRawQuery(), t.getRequestURI().getPath(), t.getRequestBody());
             
             IResponse res = this.handler.Handle(req);
 
-            String response = res.getBody();
-            byte[] bytesOut = response.getBytes("UTF-8");
+            byte[] bytesOut = res.getBodyData();
```

Comment:
App.java Line 86: Though, by this line, the Input and Output streams are already closed, it is suggested to call convenience method "close" on the HTTPExchange Object.

Reply:
I will have a look at this